### PR TITLE
gitserver: fix p4-fusion command args

### PR DESCRIPTION
**Pull request by sashaostrikov**
New code removes any -c arguments added to `p4-fusion` command, because they trigger unnecessary warnings in logs.

Test plan:
new unit test added

Closes https://github.com/sourcegraph/sourcegraph/issues/42389

### CODE REVIEW

File: cmd/gitserver/server/serverutil_test.go [hunk 1]
```diff
@@ -143,6 +143,24 @@ func TestConfigureRemoteGitCommand(t *testing.T) {
...
+	expectedEnv := []string{
+		"GIT_ASKPASS=true",
+		"GIT_SSH_COMMAND=ssh -o BatchMode=yes -o ConnectTimeout=30",
+		"GIT_HTTP_USER_AGENT=git/Sourcegraph-Bot",
+	}
+	input := exec.Command("p4-fusion", "--path", "some_path", "--client", "some_client", "--user", "some_user")
+	expectedArgs := []string{"p4-fusion", "credential.helper=", "protocol.version=2", "--path", "some_path", "--client", "some_client", "--user", "some_user"}
+
+	configureRemoteGitCommand(input, &tlsConfig{})
+	if !reflect.DeepEqual(input.Env, expectedEnv) {
```
Comment (ryanslade) [inline cmd/gitserver/server/serverutil_test.go:156]: Can you user `assert.Equal` here?

File: cmd/gitserver/server/serverutil_test.go [hunk 2]
```diff
@@ -143,6 +143,24 @@ func TestConfigureRemoteGitCommand(t *testing.T) {
...
 }
 
+func TestConfigureRemoteP4FusionCommandWithoutCArgs(t *testing.T) {
+	expectedEnv := []string{
+		"GIT_ASKPASS=true",
+		"GIT_SSH_COMMAND=ssh -o BatchMode=yes -o ConnectTimeout=30",
+		"GIT_HTTP_USER_AGENT=git/Sourcegraph-Bot",
+	}
+	input := exec.Command("p4-fusion", "--path", "some_path", "--client", "some_client", "--user", "some_user")
+	expectedArgs := []string{"p4-fusion", "credential.helper=", "protocol.version=2", "--path", "some_path", "--client", "some_client", "--user", "some_user"}
```
Comment (ryanslade) [inline cmd/gitserver/server/serverutil_test.go:153]: I don't think we expect `credential-helper` or `protocol-version`.

`-c` is used to pass configuration options to `git`. From the man page:

```
 -c <name>=<value>
           Pass a configuration parameter to the command. The value given will override values from configuration files. The <name> is expected in the same format as listed by git config (subkeys separated by dots).
```

I think we want to remove all the `-c` pairs in the extra args. So the `-c` and the next item

Comment (sashaostrikov) [reply cmd/gitserver/server/serverutil_test.go:153]: Well, that makes sense, I'll fix it now!

File: cmd/gitserver/server/serverutil_test.go (hunk 1 above)
Comment (sashaostrikov) [reply cmd/gitserver/server/serverutil_test.go]: fixed as well.

I'll make the same change to the rest of the tests in a follow-up PR to keep current diff concise

File: cmd/gitserver/server/serverutil.go [hunk 3]
```diff
@@ -239,6 +239,27 @@ func configureRemoteGitCommand(cmd *exec.Cmd, tlsConf *tlsConfig) {
 		extraArgs = append(extraArgs, "-c", "protocol.version=2")
 	}
 
+	// p4-fusion command doesn't support -c argument and passing this causes warning
+	// logs. This removes all -c arguments
+	if executable == "p4-fusion" { // lol -c proto quek -c cock -c bisch
```
Comment (ryanslade) [inline cmd/gitserver/server/serverutil.go:244]: You want to remove these comments and they may offend some people :)

Comment (sashaostrikov) [reply cmd/gitserver/server/serverutil.go]: 🤦🏻

File: cmd/gitserver/server/serverutil.go [hunk 4]
```diff
@@ -239,6 +239,27 @@ func configureRemoteGitCommand(cmd *exec.Cmd, tlsConf *tlsConfig) {
 		extraArgs = append(extraArgs, "-c", "protocol.version=2")
 	}
 
+	// p4-fusion command doesn't support -c argument and passing this causes warning
+	// logs. This removes all -c arguments
+	if executable == "p4-fusion" {
+		idx := 0
```
Comment (ryanslade) [inline cmd/gitserver/server/serverutil.go:245]: I think it would be easier to read and test this if the code that removes all the -c pairs was pulled out into it's own function. Then it would be easy to write a few tests cases where we pass args with -c and check they are removed. 

At the moment we have to run through all the `configureRemoteGitCommand` logic which makes it harder to reason about.

Comment (ryanslade) [reply cmd/gitserver/server/serverutil.go:245]: Worth testing the edge cases like empty args and uneven args

Comment (sashaostrikov) [reply cmd/gitserver/server/serverutil.go:245]: extracting into a separate function makes sense, agree. will add some tests as well, for future fail-tolerance (when new args are added to `extraArgs`)

**Review by ryanslade — APPROVED** [review]
Nice, thanks!
